homeassistant/components/eq3btsmart/number.py

```diff
@@ -0,0 +1,224 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Eq3ConfigEntry
+from .const import (
+    ENTITY_KEY_AWAY_HOURS,
+    ENTITY_KEY_AWAY_TEMPERATURE,
+    ENTITY_KEY_COMFORT,
+    ENTITY_KEY_ECO,
+    ENTITY_KEY_OFFSET,
+    ENTITY_KEY_WINDOW_OPEN_TEMPERATURE,
+    ENTITY_KEY_WINDOW_OPEN_TIMEOUT,
+    EQ3BT_STEP,
+)
+from .entity import Eq3Entity
+
+
+@dataclass(frozen=True, kw_only=True)
+class Eq3NumberEntityDescription(NumberEntityDescription):
+    """Entity description for eq3 number entities."""
+
+    value_func: Callable[[Presets], float]
+    value_set_func: Callable[
+        [Thermostat],
+        Callable[[float], Awaitable[None]],
+    ]
+    device_class: NumberDeviceClass | None = NumberDeviceClass.TEMPERATURE
+    native_unit_of_measurement: str = UnitOfTemperature.CELSIUS
+    native_min_value: float = EQ3BT_MIN_TEMP
+    native_max_value: float = EQ3BT_MAX_TEMP
+    native_step: float = EQ3BT_STEP
+    mode: NumberMode = NumberMode.BOX
+    entity_category: EntityCategory | None = EntityCategory.CONFIG
+
+
+@dataclass(frozen=True, kw_only=True)
+class Eq3RestoreNumberEntityDescription(Eq3NumberEntityDescription):
+    """Entity description for eq3 number entities that should be restored."""
+
+    value_func: Callable[[ThermostatConfig], float | int]
+
+
+NUMBER_ENTITY_DESCRIPTIONS = [
+    Eq3NumberEntityDescription(
+        key=ENTITY_KEY_COMFORT,
+        value_func=lambda presets: presets.comfort_temperature.value,
+        value_set_func=lambda thermostat: thermostat.async_configure_comfort_temperature,
+        translation_key=ENTITY_KEY_COMFORT,
+    ),
+    Eq3NumberEntityDescription(
+        key=ENTITY_KEY_ECO,
+        value_func=lambda presets: presets.eco_temperature.value,
+        value_set_func=lambda thermostat: thermostat.async_configure_eco_temperature,
+        translation_key=ENTITY_KEY_ECO,
+    ),
+    Eq3NumberEntityDescription(
+        key=ENTITY_KEY_WINDOW_OPEN_TEMPERATURE,
+        value_func=lambda presets: presets.window_open_temperature.value,
+        value_set_func=lambda thermostat: thermostat.async_configure_window_open_temperature,
+        translation_key=ENTITY_KEY_WINDOW_OPEN_TEMPERATURE,
+    ),
+    Eq3NumberEntityDescription(
+        key=ENTITY_KEY_OFFSET,
+        value_func=lambda presets: presets.offset_temperature.value,
+        value_set_func=lambda thermostat: thermostat.async_configure_temperature_offset,
+        translation_key=ENTITY_KEY_OFFSET,
+        native_min_value=EQ3BT_MIN_OFFSET,
+        native_max_value=EQ3BT_MAX_OFFSET,
+    ),
+    Eq3NumberEntityDescription(
+        key=ENTITY_KEY_WINDOW_OPEN_TIMEOUT,
+        value_set_func=lambda thermostat: thermostat.async_configure_window_open_duration,
+        value_func=lambda presets: presets.window_open_time.value.total_seconds() / 60,
+        translation_key=ENTITY_KEY_WINDOW_OPEN_TIMEOUT,
+        device_class=None,
+        native_unit_of_measurement=UnitOfTime.MINUTES,
```

Comment:
make the native unit of measurement seconds, and set `suggested_unit_of_measurement` and set that to minutes. Saves you a conversion